homeassistant/components/html5/notify.py

```diff
@@ -518,3 +531,32 @@ def _push_message(self, payload, **kwargs):
                     _LOGGER.error("Error saving registration")
                 else:
                     _LOGGER.info("Configuration saved")
+
+
+def add_jwt(timestamp, target, tag, jwt_secret):
+    """Create JWT json to put into payload."""
+    import jwt
+    jwt_exp = (datetime.fromtimestamp(timestamp) +
+               timedelta(days=JWT_VALID_DAYS))
+    jwt_claims = {'exp': jwt_exp, 'nbf': timestamp,
+                  'iat': timestamp, ATTR_TARGET: target,
+                  ATTR_TAG: tag}
+    return jwt.encode(jwt_claims, jwt_secret).decode('utf-8')
+
+
+def create_vapid_headers(vapid_email, subscription_info, vapid_private_key):
+    """Create encrypted headers to send to WebPusher."""
+    from py_vapid import Vapid
+    try:
+        from urllib.parse import urlparse
+    except ImportError:  # pragma nocover
+        from urlparse import urlparse
+    if vapid_email and vapid_private_key:
+        url = urlparse(subscription_info.get('endpoint'))
+        vapid_claims = {
+            'sub': 'mailto:{}'.format(vapid_email),
+            'aud': "{}://{}".format(url.scheme, url.netloc)
```

Comment:
Does `.netloc` include the port?

Reply:
It is copy/paste from pywebpush. According to example, no, it doesn't include port.
https://github.com/web-push-libs/pywebpush/blob/2a23f45b7819e31bd030de9fe1357a1cf7dcfdc4/pywebpush/__init__.py#L406

Reply:
It needs to since we can't assume that everyone runs on 443.

Reply:
I don't understand what to do. 
Endpoint is something browser "gives us".

Reply:
Is there any situation in which the endpoint the browser gives us could be of the form `https://hassio.local:8443` instead of `https://hassio.local`?

Reply:
I think in Chrome endpoint is always something like https://fcm.googleapis.com/fcm/send and for Firefox https://updates.push.services.mozilla.com/wpush/
I don't think it is connected with HA instance.

Reply:
Whoops, you are right, cool!